packages/api/src/webviewWindow.ts

```diff
@@ -187,46 +189,57 @@ class WebviewWindow {
    * @returns A promise resolving to a function to unlisten to the event.
    * Note that removing the listener is required if your listener goes out of scope e.g. the component is unmounted.
    */
+
   async once<T>(
     event: EventName,
     handler: EventCallback<T>
   ): Promise<UnlistenFn> {
     if (this._handleTauriEvent(event, handler)) {
       return () => {
-        // eslint-disable-next-line security/detect-object-injection
         const listeners = this.listeners[event]
-        listeners.splice(listeners.indexOf(handler), 1)
+        const idx = listeners.indexOf(handler)
+        if (idx >= 0) listeners.splice(idx, 1)
       }
     }
-    return once(event, handler, {
-      target: { kind: 'WebviewWindow', label: this.label }
-    })
+    return once(
+      event,
+      handler,
+      /**
+       * Set the window and webview background color.
+       *
+       * #### Platform-specific:
+       *
+       * - **Android / iOS:** Unsupported for the window layer.
+       * - **macOS / iOS**: Not implemented for the webview layer.
+       * - **Windows**:
+       *   - alpha channel is ignored for the window layer.
+       *   - On Windows 7, alpha channel is ignored for the webview layer.
+       *   - On Windows 8 and newer, if alpha channel is not `0`, it will be ignored.
+       *
+       * @returns A promise indicating the success or failure of the operation.
+       *
+       * @since 2.1.0
+       */ {
+        target: { kind: 'WebviewWindow', label: this.label }
+      }
+    )
   }
 
   /**
    * Set the window and webview background color.
    *
-   * #### Platform-specific:
-   *
-   * - **Android / iOS:** Unsupported for the window layer.
-   * - **macOS / iOS**: Not implemented for the webview layer.
-   * - **Windows**:
-   *   - alpha channel is ignored for the window layer.
-   *   - On Windows 7, alpha channel is ignored for the webview layer.
-   *   - On Windows 8 and newer, if alpha channel is not `0`, it will be ignored.
-   *
-   * @returns A promise indicating the success or failure of the operation.
-   *
    * @since 2.1.0
    */
   async setBackgroundColor(color: Color): Promise<void> {
-    return invoke('plugin:window|set_background_color', { color }).then(() => {
-      return invoke('plugin:webview|set_webview_background_color', { color })
-    })
+    try {
+      await invoke('plugin:window|set_background_color', { color })
+      await invoke('plugin:webview|set_webview_background_color', { color })
+    } catch (err) {
+      throw new Error(`Failed to set background color: ${String(err)}`)
```

Comment:
Pass the error as cause instead
```suggestion
      throw new Error("Failed to set background color", { cause: err })
```